neotron-bmc-pico/src/pwm.rs

```diff
@@ -0,0 +1,81 @@
+use stm32f0xx_hal::pac::{RCC, TIM14};
+
+#[derive(Debug, Default)]
+#[repr(align(8))]
+pub struct Config {
+	/// Whether the speaker is active
+	duration: u8,
+	/// The PWM ticks
+	ticks: u16,
+	/// The duty cycle (0 - 255)
+	duty_cycle: u8,
+}
+
+impl Config {
+    pub fn duty_cycle(&self) -> u8 {
+        self.duty_cycle
+    }
+
+    pub fn set_duty_cycle(&mut self, duty_cycle: u8) {
```

Comment:
If we hold the boolean 'dirty' flag internally, we would set it in each of the 'set' methods. Or if we own `TIM14` we could stuff the value directly in the hardware (as well as caching it locally).